lib/src/Uncertainty/Algorithm/Simulation/PhysicalSpaceCrossEntropyImportanceSampling.cxx

```diff
@@ -0,0 +1,230 @@
+
+    if (activeParameters_.getSize() == 0)
+    throw InvalidArgumentException(HERE) << "In PhysicalSpaceCrossEntropyImportanceSampling::PhysicalSpaceCrossEntropyImportanceSampling, active parameters size (" << activeParameters_.getSize() << " has to be greater than zero";    
+    
+    
+    for (UnsignedInteger i = 0; i < activeParameters_.getSize(); ++i)
+    {
+      parameters[activeParameters_[i]] = initialAuxiliaryDistributionParameters[i];
+    }      
+    auxiliaryDistribution_.setParameter(parameters);
+  }
+
+
+/* Virtual constructor */
+PhysicalSpaceCrossEntropyImportanceSampling * PhysicalSpaceCrossEntropyImportanceSampling::clone() const
+{
+  return new PhysicalSpaceCrossEntropyImportanceSampling(*this);
+}
+
+namespace {
+class KullbackLeiblerDivergenceObjective : public EvaluationImplementation
+{
+public:
+  KullbackLeiblerDivergenceObjective(const Sample auxiliaryCriticInputSample,
+                                     const Point & initialCriticInputSamplePDFValue,
+                                     const Distribution & auxiliaryDistribution,
+                                     const Indices & activeParameters,
+                                     const Scalar numberSamples)
+  : EvaluationImplementation()
+  , auxiliaryCriticInputSample_(auxiliaryCriticInputSample)
+  , activeParameters_(activeParameters)
+  , auxiliaryDistribution_(auxiliaryDistribution)
+  , initialCriticInputSamplePDFValue_(initialCriticInputSamplePDFValue)
+  , numberSamples_(numberSamples)
+  {
+    // Nothing to do
+  }
+  
+  KullbackLeiblerDivergenceObjective * clone () const override
+  {
+    return new KullbackLeiblerDivergenceObjective(*this);
+  }
+  
+  Point operator()(const Point & x) const override
+  {
+    Distribution distrib = auxiliaryDistribution_;
+  
+    // update auxiliary distribution
+    Point parameters(auxiliaryDistribution_.getParameter());
+  
+    for (UnsignedInteger i = 0; i < activeParameters_.getSize(); ++i)
+    {
+      parameters[activeParameters_[i]] = x[i];
+    }
+    distrib.setParameter(parameters);
+  
+    Point criticSamplesAuxiliaryPDFValue = distrib.computePDF(auxiliaryCriticInputSample_).asPoint();
+    Point criticSamplesAuxiliaryLogPDFValue = distrib.computeLogPDF(auxiliaryCriticInputSample_).asPoint();
+  
+    Scalar objectiveFunction = 0.0;
+  
+    for(UnsignedInteger j = 0; j < auxiliaryCriticInputSample_.getSize(); ++j)
+    {  
+      objectiveFunction += initialCriticInputSamplePDFValue_[j]/criticSamplesAuxiliaryPDFValue[j]*criticSamplesAuxiliaryLogPDFValue[j];
+    }
+    objectiveFunction = 1/numberSamples_*objectiveFunction;
+  
+    return Point(1,objectiveFunction);
+  }
+  
+  UnsignedInteger getInputDimension() const override
+  {
+    return activeParameters_.getSize();
+  }
+  
+  UnsignedInteger getOutputDimension() const override
+  {
+    return 1;
+  }
+  
+  
+protected :
+Sample auxiliaryCriticInputSample_;
```

Comment:
I think that the whole block should be indented.

Reply:
fixed